sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCScanBuilder.scala

```diff
@@ -126,24 +126,23 @@ case class JDBCScanBuilder(
       upperBound: Double,
       withReplacement: Boolean,
       seed: Long): Boolean = {
-    if (jdbcOptions.pushDownTableSample &&
-      JdbcDialects.get(jdbcOptions.url).supportsTableSample) {
+    if (jdbcOptions.pushDownTableSample && dialect.supportsTableSample) {
```

Comment:
shall we make two more changes?
1. remove the JDBC option for pushdown (false by default AFAIK)
2. jdbc dialect should return false in these new pushdown APIs, to keep the old false by default behavior.

Reply:
OK. Let's do it in next PR.